packages/core/src/Cardano/types/Asset.ts

```diff
@@ -14,7 +14,7 @@ export const AssetName = (value: string): AssetName => {
       throw new InvalidStringError('too long');
     }
   }
-  return value as unknown as AssetName;
+  return value.toLowerCase() as unknown as AssetName;
```

Comment:
These opaque string types encapsulated a byte array as a hex string, however, we need to be careful when comparing with these since a HEX string is case INsensitive, and can have valid representations in upper or lower case characters. We must either make sure they are in the same case all the time (byte converting them to lower or upper case) or better yet, store this as byte arrays (since this is essentially what they are) and make byte to byte comparisons. I understand the convenience of having those as strings instead of byte arrays, however, I think when we use them, we always need to convert them to byte arrays (pass down to serialization lib).

I think we may as well just keep them as byte arrays internally from the start (keeping the string constructor) and just add a toString() method that converts back to its hexadecimal representation.

Reply:
> I think we may as well just keep them as byte arrays internally from the start

Agree

Reply:
Nice catch. We should probably apply this change to all opaque strings that are hex strings. Byte array does seem like a nice way to store it internally, but I'm not 100% sold on it because it would make it more difficult to debug: byte strings are not very human readable. Maybe it could work if if the type had a hex string getter.